src/tv2-common/helpers/graphics/pilot/index.ts

```diff
@@ -70,33 +77,33 @@ export function CreatePilotGraphic(
 		return
 	}
 
-	const engine = parsedCue.target
-
 	if (IsTargetingFull(engine)) {
-		actions.push(
-			CreatePilotAdLibAction(config, context, parsedCue, engine, settings, adlib, adlibRank, externalSegmentId)
-		)
+		actions.push(CreatePilotAdLibAction(pilotGraphicProps))
+	}
+
+	if (!(IsTargetingOVL(pilotGraphicProps.engine) && adlib)) {
+		pieces.push(CreatePilotPiece(pilotGraphicProps))
 	}
 
-	if (!(IsTargetingOVL(engine) && adlib)) {
-		pieces.push(CreateFullPiece(config, context, partId, parsedCue, engine, settings, adlib, externalSegmentId))
+	if (IsTargetingOVL(engine) && adlib) {
+		adlibPieces.push(CreatePilotAdlibPiece(pilotGraphicProps))
 	}
 
 	if (IsTargetingFull(engine)) {
-		pieces.push(CreateFullDataStore(config, context, settings, parsedCue, engine, partId, adlib, externalSegmentId))
+		pieces.push(CreateFullDataStore(pilotGraphicProps))
 	}
 }
 
-function CreatePilotAdLibAction(
-	config: TV2BlueprintConfig,
-	context: IShowStyleUserContext,
-	parsedCue: CueDefinitionGraphic<GraphicPilot>,
-	engine: GraphicEngine,
-	settings: PilotGeneratorSettings,
-	adlib: boolean,
-	adlibRank: number,
-	segmentExternalId: string
-) {
+function CreatePilotAdLibAction({
+	config,
+	context,
+	parsedCue,
+	engine,
+	settings,
+	adlib,
+	adlibRank,
+	segmentExternalId
+}: PilotGraphicProps) {
```

Comment:
Readability of the body. I could destructure in the first line instead, but this looks good to me.

Reply:
> Readability of the body

Not sure I entirely follow what you mean here. Are you referring to the body of the method or the "body" of the paramater?

For the parameter: It's way less clutter to look at `someFunctionName(pilotGraphicProps: PilotGraphicProps)` than having to read multiple lines of code.

For the body of the method: With this syntax we are just going to have a variable "floating" that we need to mentally remember belongs to a PilotGraphicProps. In a long method (which sadly there are a lot of in Sofie) then this mental mapping will get trickier the longer you get down in the method (because there is so much you need to be aware of in long methods). By declaring the parameter as `pilotGraphicProps: PilotGraphicProps` you don't have to do that mental mapping because the code outright does it for you by telling you that this attribute belongs to a piloGraphicProps. 

```
interface Car {
  amountOfWheels: number
}

void myFunction({amountOfWheels}: Car) {
  // 50+ lines of code
  if (amountOfWheels > 4) { // Do something) }
}

void myFunction(car: Car) {
  // 50+ lines of code
  if (car.amountOfWheels > 4) { // Do something) }
}
```
Bonus: You also don't have to type out all the attributes of the interface everytime you want to use the interface.


Sidenote: Typescript is an object-oriented language, so lets write object-oriented code. (A thing that we barely utilize in the Sofie codebase which is a real shame...)